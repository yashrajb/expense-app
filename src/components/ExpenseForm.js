import React from 'react';
import Moment from 'moment';
import { SingleDatePicker } from 'react-dates';
//uimport 'react-dates/lib/css/_datepicker.css';

// console.log(Moment());

class ExpenseForm extends React.Component {

constructor(props) {

    super(props);

this.state = {
    description: props.expenses ? props.expenses.description : '',
    amount:props.expenses ? (props.expenses.amount / 100).toString() : undefined,
    note:props.expenses ? props.expenses.note : '',
    createdAt:props.expenses ? Moment(props.expenses.createdAt) : Moment(),
    focus:false,
    error:''

};


}

   

    onChangeDescription = (e) => {

        const description = e.target.value;
        this.setState({
            description
        })


    }

    onChangeAmount = (e) => {
        
            const amount = e.target.value;
            if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {

                this.setState({
                    amount
                });
        
        
            }

    }
                

    onChangeNote = (e) => {
                
        
        const note = e.target.value;
        this.setState(()=>({
         note
        }));
                
                
                    }

        onDateChange = (createdAt) => {

            this.setState({

                createdAt

            });

        }

        onFocusedChange = ({ focused }) => {
            this.setState({
                focus:focused
            });
        }

        onSubmit = (e) => {

            e.preventDefault();

            if(!this.state.description || !this.state.amount) {

                this.setState({
                    error:"Please fill the description or amount."
                });

            }else {

                this.setState({
                    error:''
                });

                this.props.onSubmit({
                    description:this.state.description,
                    createdAt:this.state.createdAt.valueOf(),
                    amount:parseFloat(this.state.amount, 10) * 100,
                    note:this.state.note
                });

                
                
            }



        }

   

 render() {


    return (

        <div>
        { (this.state.error === " ") ? '' :<p>{this.state.error}</p>  }
        <h1>Expense Form</h1>
        <form onSubmit={this.onSubmit}>
        <input type="text" 
        value={this.state.description} autoFocus 
        onChange={this.onChangeDescription} placeholder="Description...."/>

        <input
        type="text"
        value={this.state.amount}
        onChange={this.onChangeAmount}
        placeholder="Amount..." />
 
        <SingleDatePicker
        date={this.state.createdAt} // momentPropTypes.momentObj or null
        onDateChange={this.onDateChange} // PropTypes.func.isRequired
        focused={this.state.focus} // PropTypes.bool
        numberOfMonths={1}
        isOutsideRange={() => false}
        onFocusChange={this.onFocusedChange} // PropTypes.func.isRequired
      />
    

        <textarea 
        type="text"
        value={this.state.note} 
        onChange={this.onChangeNote}
        placeholder="Note..."

        />
       

        <input type="submit"
        value="Submit"
        />

        </form>
        


        
        </div>




    );




 }




}

export default ExpenseForm;