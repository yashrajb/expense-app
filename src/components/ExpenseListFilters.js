import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount,setStartDate,setEndDate } from '../actions/filters';
import {DateRangePicker} from 'react-dates';
import Moment from 'moment';


class ExpenseListFilters extends React.Component {

 

   state = {
      callendarFocused:null
    }



onDatesChange = ({ startDate, endDate }) => {

  this.props.setStartDate(startDate);
  this.props.setEndDate(endDate);

  

}

onTextChange = (e) => {
  this.props.setTextFilter(e.target.value);
}

onSelectChange = (e) => {
  if (e.target.value === 'date') {
   this.props.sortByDate();
  } else if (e.target.value === 'amount') {
    this.props.sortByAmount();
  }
}
focusedInput = (callendarFocused) => {
      this.setState({ callendarFocused });
}
render() {


return (

  <div>
  <input
    type="text"
    value={this.props.filters.text}
    onChange={this.onTextChange}
  />
  <select
    value={this.props.filters.sortBy}
    onChange={this.onSelectChange}
  >
    <option value="date">Date</option>
    <option value="amount">Amount</option>
  </select>

  <DateRangePicker
  startDate={this.props.filters.startDate} 
  endDate={this.props.filters.endDate} 
  onDatesChange={this.onDatesChange} 
  focusedInput={this.state.callendarFocused} 
  onFocusChange={this.focusedInput} 
  numberOfMonths={1}
  isOutsideRange={() => false}
  hideKeyboardShortcutsPanel={true}
  />


</div>


);





  }
 



}



const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({

  setStartDate: (startDate) => { 
    dispatch(setStartDate(startDate));
    },
  setEndDate: (endDate) => {
    dispatch(setEndDate(endDate));
  },
  setTextFilter: (e) => { dispatch(setTextFilter(e)); },
  sortByDate: () => { dispatch(sortByDate()); },
  sortByAmount: () => { dispatch(sortByAmount()); }


});

export default connect(mapStateToProps)(ExpenseListFilters);

export { ExpenseListFilters };
