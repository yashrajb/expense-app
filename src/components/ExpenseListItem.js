import React from 'react';
import { Link } from 'react-router-dom';


const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
  <div>
    <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
    <p>{amount} - {createdAt}</p>
  </div>
);

export default ExpenseListItem;
