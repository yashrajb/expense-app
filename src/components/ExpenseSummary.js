import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expenseTotal from './expense-total';
import numeral from 'numeral';

const expenseCount = (props) => {
    
     

     return (<p>viewing {props.state.length} expenses and totat is {numeral(expenseTotal(props.state) / 100).format('$0,0.00')}</p>);
}


const mapStateToProps = (state) => ({
    state:selectExpenses(state.expenses,state.filters)
  })


export default connect(mapStateToProps)(expenseCount)