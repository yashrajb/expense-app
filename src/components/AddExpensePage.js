import React from 'react';
import ExpenseForm from './ExpenseForm.js';
import { addExpense } from '../actions/expenses.js';
import { connect } from 'react-redux';




class AddExpensePage extends React.Component {

  onSubmit = (expense) => {
    this.props.addExpense(expense);
    this.props.history.push('/');
  }

      render() {


              return (
                <div>
                <ExpenseForm onSubmit={this.onSubmit}/>
                </div>
              );


      };


}

const mapDispatchToProps = (dispatch) => {
  return {
    addExpense:(expense) => dispatch(addExpense(expense))
  }
} 
export default connect(undefined,mapDispatchToProps)(AddExpensePage);
export { AddExpensePage };