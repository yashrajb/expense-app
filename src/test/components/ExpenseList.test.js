import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixture/commonData';
import ExpenseListItem from '../../components/ExpenseListItem';

test('should render expenselistitem',() => {
    const result = shallow(<ExpenseList expenses={expenses}/>);
    expect(result).toMatchSnapshot();
});

test('should render expenselistitem',() => {
    const result = shallow(<ExpenseList expenses={[]}/>);
    expect(result).toMatchSnapshot();
});




