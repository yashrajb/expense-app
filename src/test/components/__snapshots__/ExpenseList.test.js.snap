// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenselistitem 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id="123456789"
    key="123456789"
    note=""
  />
  <ExpenseListItem
    amount={19500}
    createdAt={-345600000}
    description="Rent"
    id="123456789abcdefgh"
    key="123456789abcdefgh"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="Credit card"
    id="123456789abcdefghijkl"
    key="123456789abcdefghijkl"
    note=""
  />
</div>
`;

exports[`should render expenselistitem 2`] = `
<div>
  <h1>
    Expense List
  </h1>
  <p>
    No expenses
  </p>
</div>
`;
